rubin_scheduler/scheduler/basis_functions/feasibility_funcs.py

```diff
@@ -57,6 +61,84 @@ def check_feasibility(self, conditions):
         return result
 
 
+class InSeasonBasisFunction(BaseBasisFunction):
+    """Only let a survey go if it is in a defined season
+
+    Parameters
+    ----------
+    seasons : `list`
+        List of mjd pairs. feasability will only pass if the
+        current MJD falls in a viable range, e.g.
+        seaons=[[1,2], [10,13]], means observations can happen
+        on mjd=1,2,10,11,12,13.
+    """
+
+    def __init__(self, seasons=[]):
+        super().__init__()
+        self.seasons = seasons
+
+    def check_feasibility(self, conditions):
+        result = False
+        for season_range in self.seasons:
+            if np.min(season_range) <= conditions.mjd <= np.max(season_range):
+                result = True
+        return result
+
+
+class AirmassPointRangeBasisFunction(BaseBasisFunction):
```

Comment:
It is a feasibility function and a basis function, because all feasibility functions are subclasses of basis functions.
Right now I'm using this with the Roman observations to give them an airmass limit.